frontend/src/app/main/data/workspace/selection.cljs

```diff
@@ -184,24 +198,51 @@
                  (rx/map select-shapes))))))))
 
 (defn select-inside-group
-  [group-id position]
-  (ptk/reify ::select-inside-group
-    ptk/WatchEvent
-    (watch [_ state stream]
-      (let [page-id  (:current-page-id state)
-            objects  (dwc/lookup-page-objects state page-id)
-            group    (get objects group-id)
-            children (map #(get objects %) (:shapes group))
-
-            ;; We need to reverse the children because if two children
-            ;; overlap we want to select the one that's over (and it's
-            ;; in the later vector position
-            selected (->> children
-                          reverse
-                          (d/seek #(geom/has-point? % position)))]
-        (when selected
-          (rx/of (deselect-all) (select-shape (:id selected))))))))
-
+  ([group-id position] (select-inside-group group-id position false))
+  ([group-id position deep-children]
+   (ptk/reify ::select-inside-group
+     ptk/WatchEvent
+     (watch [_ state stream]
+       (let [page-id  (:current-page-id state)
+             objects  (dwc/lookup-page-objects state page-id)
+             group    (get objects group-id)
+             children (map #(get objects %) (:shapes group))
+
+             ;; We need to reverse the children because if two children
+             ;; overlap we want to select the one that's over (and it's
+             ;; in the later vector position
+             selected (->> children
+                           reverse
+                           (d/seek #(geom/has-point? % position)))]
+         (when selected
+           (rx/of (deselect-all) (select-shape (:id selected)))))))))
+
+(defn select-last-layer
+  ([position]
+   (ptk/reify ::select-last-layer
+     ptk/WatchEvent
+     (watch [_ state stream]
+       (let [page-id  (:current-page-id state)
+             objects (dwc/lookup-page-objects state page-id)
+             find-shape
+             (fn [selection]
+               (when (seq selection)
```

Comment:
minor: probably this check can be done in the rx stream transformations; maybe with `(rx/filter #(boolean (seq %))` before the `rx/map`.